### DISCUSSION TITLE
Can I pass options into wagmi connectors?

### DISCUSSION THREAD

**Discussion by frolic** [rainbow-me/rainbowkit#616] [Q&A] [answered]
I want to enable this: https://wagmi.sh/docs/connectors/metaMask#unstable_shimonconnectselectaccount

But I also want to be able to use RainbowKit's `getDefaultWallets` (I think I _have_ to use this for RainbowKit compatibility?). It looks like it's not easy to pass in options to wagmi's connector under the hood, since RainbowKit wraps the connectors in a few layers of abstraction:
https://github.com/rainbow-me/rainbowkit/blob/239a0ff765084aa827b48df58f1c37464da3ae80/packages/rainbowkit/src/wallets/walletConnectors/metaMask/metaMask.ts#L61-L64

I tried passing a custom list of `connectors` to wagmi's `createClient`, where I'd override the MetaMask one but with the option toggled on, but no luck there (MetaMask as an option disappears from the RainbowKit UI).

**Comment by sicktastic** [comment]
I am facing the same issue trying to connect [wagmi connector with safe](https://github.com/safe-global/safe-apps-sdk/tree/master/packages/safe-apps-wagmi).

Need to do something like this:

```javascript
const client = createClient({
  connectors: [
    new SafeConnector({ chains }),
    new MetaMaskConnector({ chains }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
    new InjectedConnector({
      chains,
      options: {
        name: 'Injected',
        shimDisconnect: true,
      },
    }),
  ],
  provider,
  webSocketProvider,
});
```

I can't seem to make with work with the `getDefaultWallets` from 🌈

**Comment by cryptoKevinL** [comment]
any updates here?  We are hoping to pass in the address for the connector, so we can ask a specific account to sign a message, without having to have the user connect manually?

**Comment by DanielSinclair** [comment, marked as answer]
@cryptoKevinL @holic This is now supported across all of the connectors. The connectors are all typed and look like this, with `walletConnectOptions` supporting any WalletConnect v1 or v2 options, and overflow `options` supporting the Wagmi `InjectedConnectorOptions` or in this case the subclass `MetaMaskConnectorOptions`. You can reference our connectors [here](https://github.com/rainbow-me/rainbowkit/tree/main/packages/rainbowkit/src/wallets/walletConnectors)
```ts
export interface MetaMaskWalletLegacyOptions {
  projectId?: string;
  chains: Chain[];
  walletConnectVersion: '1';
  walletConnectOptions?: WalletConnectLegacyConnectorOptions;
}

export interface MetaMaskWalletOptions {
  projectId: string;
  chains: Chain[];
  walletConnectVersion?: '2';
  walletConnectOptions?: WalletConnectConnectorOptions;
}

export const metaMaskWallet = ({
  chains,
  projectId,
  walletConnectOptions,
  walletConnectVersion = '2',
  ...options
}: (MetaMaskWalletLegacyOptions | MetaMaskWalletOptions) & MetaMaskConnectorOptions): Wallet
  ```
